src/test/java/org/fundacionjala/coding/yerel/KataBankOCRTest.java

```diff
@@ -0,0 +1,140 @@
+                + "|_||_||_||_||_||_||_||_||_|\n";
+        assertEquals("000000000", kataBankOCR.accountNumber(number));
+    }
+
+    /**
+     * test case for validate 333333333.
+     */
+    @Test
+    public void accountNumberTwo() {
+        String number = " _  _  _  _  _  _  _  _  _ \n"
+                + " _| _| _| _| _| _| _| _| _|\n"
+                + " _| _| _| _| _| _| _| _| _|\n";
+        assertEquals("333333333", kataBankOCR.accountNumber(number));
+    }
+
+    /**
+     * test case for validate 123456789.
+     */
+    @Test
+    public void accountNumberThree() {
+        String number = "    _  _     _  _  _  _  _ \n"
+                      + "  | _| _||_||_ |_   ||_||_|\n"
+                      + "  ||_  _|  | _||_|  ||_| _|\n";
+        assertEquals("123456789", kataBankOCR.accountNumber(number));
+    }
+
+    /**
+     * test case for validate number 000000000.
+     */
+    @Test
+    public void convertNumber() {
+        List<StringBuilder> listNumber = new ArrayList<>();
+        String numberZero = " _ | ||_|";
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        assertEquals("000000000", kataBankOCR.convertNumber(listNumber));
+    }
+
+    /**
+     * test validate a digit is no read.
+     */
+    @Test
+    public void convertNumberTwo() {
+        List<StringBuilder> listNumber = new ArrayList<>();
+        String unknown = " _ |_| _ ";
+        String numberZero = " _ | ||_|";
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(numberZero));
+        listNumber.add(new StringBuilder(unknown));
+        assertEquals("00000000?", kataBankOCR.convertNumber(listNumber));
+    }
+
+    /**
+     *
+     */
+    @Test
+    public void checksumValidate() {
+        assertFalse(kataBankOCR.checksumValidate("664371495"));
+    }
+
+    /**
+     * test case validate checksum 86110??36.
+     */
+    @Test
+    public void checksumValidateTwo() {
+        assertTrue(kataBankOCR.checksumValidate(ACCOUNT));
+    }
+
+    /**
+     * test case validate checksum 457508000.
+     */
+    @Test
+    public void seeResultOne() {
+        String expecteResult = "457508000";
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use already-defined constant 'ACCOUNT' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)